homeassistant/components/devolo_home_control/config_flow.py

```diff
@@ -29,22 +34,35 @@ def __init__(self):
     async def async_step_user(self, user_input=None):
         """Handle a flow initiated by the user."""
         if self.show_advanced_options:
-            self.data_schema = {
-                vol.Required(CONF_USERNAME): str,
-                vol.Required(CONF_PASSWORD): str,
-                vol.Required(CONF_MYDEVOLO, default=DEFAULT_MYDEVOLO): str,
-            }
+            self.data_schema.update(
+                {
+                    vol.Required(CONF_MYDEVOLO, default=DEFAULT_MYDEVOLO): str,
+                }
+            )
         if user_input is None:
             return self._show_form(user_input)
-        user = user_input[CONF_USERNAME]
-        password = user_input[CONF_PASSWORD]
-        mydevolo = Mydevolo()
-        mydevolo.user = user
-        mydevolo.password = password
-        if self.show_advanced_options:
-            mydevolo.url = user_input[CONF_MYDEVOLO]
-        else:
-            mydevolo.url = DEFAULT_MYDEVOLO
+        return await self._connect_mydevolo(user_input)
+
+    async def async_step_zeroconf(self, discovery_info: DiscoveryInfoType):
+        """Handle zeroconf discovery."""
+        try:
+            # Check if it is a gateway
+            if discovery_info["properties"]["MT"] in ("2600", "2601"):
```

Comment:
What is 2600, 2601? Can it be made into constant so it's easier to follow?

Reply:
Other devolo devices react to the same zeroconf domain, but be can only handle models having that property in this integration. I moved it to a constant to make the purpose clear.